### PR TITLE
Added colocate gradients argument to optimize_loss

### PR DESCRIPTION
**Pull request by alexgkendall**
This PR adds a colocate gradients argument to optimize_loss which is useful for multi GPU training where the model is split across multiple GPUs.

### CODE REVIEW

**Comment by alexgkendall** [comment]
I signed it!

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Review by martinwicke — APPROVED** [review]

File: tensorflow/contrib/layers/python/layers/optimizers.py
Comment (vrv) [inline tensorflow/contrib/layers/python/layers/optimizers.py]: maybe add documentation? also maybe name colocate_gradients_with_ops to match?

Comment (martinwicke) [inline tensorflow/contrib/layers/python/layers/optimizers.py]: Yes please. <shame, ding ding>

**Review by martinwicke — CHANGES_REQUESTED** [review]

File: tensorflow/contrib/layers/python/layers/optimizers.py
Comment (alexgkendall) [inline tensorflow/contrib/layers/python/layers/optimizers.py]: Done.

**Review by martinwicke — APPROVED** [review]

**Comment by martinwicke** [comment]
Jenkins, test this please.

**Comment by alexgkendall** [comment]
@martinwicke do you know what is causing these tests to fail?

**Comment by martinwicke** [comment]
The Linux/Py3 failure is unrelated. 

The Mac failure was a network problem, and should be just a flake. I re-ran it, with results eventually showing up at 
http://ci.tensorflow.org/job/tensorflow-pull-requests-mac/2023

Once that's clean we can merge. @jhseu, can you merge this if the above test comes back clean?

**Comment by jhseu** [comment]
Yeah. The mac flakes are fixed internally, so I'll merge this and rerun the tests once I push the internal changes.

**Pull request merged**
